spring-pulsar/src/test/java/org/springframework/pulsar/core/PulsarProducerFactoryTests.java

```diff
@@ -120,4 +145,29 @@ protected void assertProducerHasTopicSchemaAndRouter(Producer<String> producer,
 	protected abstract PulsarProducerFactory<String> producerFactory(PulsarClient pulsarClient,
 			Map<String, Object> producerConfig);
 
+	@SuppressWarnings("rawtypes")
+	private static class MockInterceptor implements ProducerInterceptor {
```

Comment:
Any reason to not just use `Mockito.mock`?

Reply:
I added this before I saw that we are using Mockito in other places. I'll swap it out